GameData/RP-1/Contracts/Commercial Applications 1/EarlyNavSat-CA.cfg

```diff
@@ -0,0 +1,158 @@
+	genericDescription = Put a satellite with the required amount of navigation satellite payload into the desired orbit.
+
+	synopsis = Launch a Navigation Satellite for a customer.
+
+	completedMessage = Success! The navigation satellite was launched into the proper orbit.  Precision satellite-based navigation is one step closer to being a reality.
+
+	sortKey = 499
+
+	cancellable = true
+	declinable = true
+	autoAccept = false
+	minExpiry = 1
+	maxExpiry = 1
+	maxCompletions = 6
+	maxSimultaneous = 1
+	deadline = 0
+
+	targetBody = HomeWorld()
+
+
+	// ************ REWARDS ************
+	prestige = Trivial       // 1.0x
+	advanceFunds = 0
+	rewardFunds = 0
+	rewardScience = 0
+	rewardReputation = 20
+	failureReputation = 0 // was @rewardReputation
+	failureFunds = 0
+
+	// ************ REQUIREMENTS ************
+
+	REQUIREMENT
+	{
+			name = ProgramActive
+			type = ProgramActive
+			program = EarlyCommApp
+	}
+	
+	REQUIREMENT
+	{
+		name = CompleteContract
+		type = CompleteContract
+		contractType = FirstNavSat-CA
+		title = Complete First Navigation Satellite contract
+	}
+
+	DATA
+	{
+		type = int
+		index = $EarlyNavSat_Count + 0
+	}
+
+	BEHAVIOUR
+	{
+		name = IncrementTheCount
+		type = Expression
+
+		EarlyNavSat_Count = $EarlyNavSat_Count + 1
+	}
+
+	DATA
+	{
+		type = float
+		payload = 200
+	}
+
+	// ************ PARAMETERS ************
+
+	PARAMETER
+	{
+		name = EarlyNavSat
+		type = VesselParameterGroup
+		define = EarlyNavSatellite
+		dissassociateVesselsOnContractCompletion = true
+		title = Navigation Satellite (Early)
+
+		PARAMETER
+		{
+			name = NewVessel
+			type = NewVessel
+			title = Launch a new vessel
+			hideChildren = true
+		}
+		PARAMETER
+		{
+			name = Crewmembers
+			type = HasCrew
+			minCrew = 0
+			maxCrew = 0
+			title = Uncrewed
+			hideChildren = true
+		}
+		PARAMETER
+		{
+			name = HasNavSatPayload
+			type = HasResource
+			resource = NavSatPayload
+			minQuantity = @/payload-0.1
+			title = Have a NavSatPayload of at least @/payload units on the craft
+			hideChildren = true
+		}
+		PARAMETER
+		{
+			name = Orbit
+			type = Orbit
+			minInclination = Round(Random(30, 90))
+			minPeA = Round(Random(300000, 400000), 1000)
+			minApA = Round(Random(500000, 1000000), 1000)
+			maxApA = Round(Random(1000000, 3000000), 1000)
```

Comment:
Will it accept functions?  Can I do minApA + 50000 or just simplify it at 1050000?

Reply:
> Will it accept functions? Can I do minApA + 50000 or just simplify it at 1050000?

Not 100% in the case above, though seems to be possible according to [the CC wiki for Expressions](https://github.com/jrossignol/ContractConfigurator/wiki/Expressions) Might be easier to put those values into `DATA` nodes and use an @. 

Example from OrbitFlight2.cfg:
`endApA = @startApA + Round(Random(150000, 800000), 25000)`

Reply:
I believe you can also self-reference there if you like, because certainly we do it all the time in `title` blocks where we reference a value in the node itself. But @Teykn 's approach is I think cleaner since it puts all the 'mission-critical' variables at the top in one place, rather than strewn throughout the parameters.